xml_converter/generators/code_generator.py

```diff
@@ -527,6 +530,10 @@ def generate_cpp_variable_data(
                 else:
                     uses_file_path = False
 
+                if "side_effects" in fieldval:
+                    for side_effect in fieldval['side_effects']:
+                        side_effects.append(lowercase(side_effect, "_"))
```

Comment:
Something seems strange with using `lowercase(side_effect, "_")` here. I think it is technically correct, but something feels odd about it, like there should be a generic function for "turn a markdown doc name into an html class variable name". Is it possible that we had only ever needed to perform that sort of action in one place before and now it is required in a different place?
Either way it should probably be generalized so that 1) it is clear why you are lowercasing this, and 2) if the convention of converting a documentation name to a variable name is ever changed elsewhere this is also easily discovered.

Reply:
One of the tests we will need to write is one that has a markdown file with very different values for everything. Then validate we are getting the values from the right places. Because now several values are very similar and we might be pulling things from incorrect locations. But we can save this for much later, but before the total project is complete.